src/Subscriptions/Repositories/SubscriptionRepository.php

```diff
@@ -256,6 +258,21 @@ public function delete(Subscription $subscription): bool
         return true;
     }
 
+    /**
+     * @param int $subscriptionId
+     * @param int $donationId
+     *
+     * @return void
+     */
+    public function updateLegacyParentPaymentId(int $subscriptionId, int $donationId)
+    {
+        DB::table('give_subscriptions')
+            ->where('id', $subscriptionId)
+            ->update([
+                'parent_payment_id' => $donationId,
+            ]);
+    }
```

Comment:
Since this is a public method on the repository, we should add some validation. I'm thinking at the very least make sure the donation exists since the where clause would fail if the subscriptionId does not exist. Also wrapping in a transaction wouldn't hurt.

Reply:
I added further documentation in 515d11cbd.

I'm not really sure any further validation is necessary. It's not like we check to see if the subscription exists whenever we save a donation with a `subscriptionId`. The where clause won't fail if the subscription doesn't exist, it will simply do nothing.

Also, a transaction wouldn't do anything here. If it fails there's nothing to rollback.

Reply:
Okay thanks, it might be beneficial to actually add that validation when saving a donation (if a subscription exists) - we _are_ validating that the donor exists.  Can discuss another time.